apps/admin/frontend/src/components/export_batch_tally_results_button.tsx

```diff
@@ -1,18 +1,35 @@
 import React, { useContext, useState } from 'react';
 import { Button } from '@votingworks/ui';
 import { generateBatchResultsDefaultFilename } from '@votingworks/utils';
-import { assert } from '@votingworks/basics';
-import { SaveFileToUsb, FileType } from './save_file_to_usb';
+import { assert, err } from '@votingworks/basics';
 import { AppContext } from '../contexts/app_context';
-import { generateBatchTallyResultsCsv } from '../utils/generate_batch_tally_results_csv';
-import { getCastVoteRecordFileMode } from '../api';
+import { exportBatchResults, getCastVoteRecordFileMode } from '../api';
+import {
+  SaveBackendFileModal,
+  SaveBackendFileResult,
+} from './save_backend_file_modal';
 
 export function ExportBatchTallyResultsButton(): JSX.Element {
   const [isSaveModalOpen, setIsSaveModalOpen] = useState(false);
-  const { fullElectionTally, electionDefinition } = useContext(AppContext);
+  const { electionDefinition } = useContext(AppContext);
   assert(electionDefinition);
   const { election } = electionDefinition;
 
+  const exportBatchResultsMutation = exportBatchResults.useMutation();
+
+  async function onSave(path: string): Promise<SaveBackendFileResult> {
+    try {
+      const exportResult = await exportBatchResultsMutation.mutateAsync({
+        path,
+      });
+      return exportResult;
+    } catch (error) {
+      // Handled by default query client error handling
```

Comment:
If it's handled upstream, do we even need a `catch` here?

Reply:
Great question. To be honest, I do not understand the react query error handling, but this is how I've learned from @arsalansufi and @jonahkagan to handle errors in cases where I have to do `async` mutations. I know it's clunky. See the [rearchitecture presentation](https://docs.google.com/presentation/d/1qcohhkeavCVjTZAAhGbbB7sbnvDJLlNov3OTLijfcvs/edit#slide=id.g1f8414d696e_0_361).

Reply:
https://github.com/votingworks/vxsuite/pull/3573/commits/bc7247235488518bb5a6c533678b585bdcb2929c